File: CineFinder.API/Models/Avaliacao.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CineFinder.API.Models;

public class Avaliacao
{
    public int Id { get; set; }

    [Required]
    public int UsuarioId { get; set; }

    [ForeignKey(nameof(UsuarioId))]
    public Usuario? Usuario { get; set; }

    public int? FilmeId { get; set; }

    [ForeignKey(nameof(FilmeId))]
    public Filme? Filme { get; set; }

    public int? SerieId { get; set; }

    [ForeignKey(nameof(SerieId))]
    public Serie? Serie { get; set; }

    [Required, Range(1, 10)]
    public int Nota { get; set; }

    [MaxLength(1000)]
    public string? Comentario { get; set; }

    public DateTime DataCriacao { get; set; } = DateTime.UtcNow;
}

File: cinefinder-java/src/main/java/com/cinefinder/model/Avaliacao.java
package com.cinefinder.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity @Table(name = "avaliacoes")
public class Avaliacao {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY) @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    /** ID do filme na API C# / TMDB (referência externa, sem FK local) */
    @Column(name = "filme_id")
    private Long filmeId;

    /** ID da série na API C# / TMDB (referência externa, sem FK local) */
    @Column(name = "serie_id")
    private Long serieId;

    @Column(nullable = false)
    private Integer nota; // 1–10

    @Column(columnDefinition = "TEXT")
    private String comentario;

    @Column(nullable = false)
    private LocalDateTime dataCriacao;

    public Avaliacao() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }

    public Long getFilmeId() {
        return filmeId;
    }

    public void setFilmeId(Long filmeId) {
        this.filmeId = filmeId;
    }

    public Long getSerieId() {
        return serieId;
    }

    public void setSerieId(Long serieId) {
        this.serieId = serieId;
    }

    public Integer getNota() {
        return nota;
    }

    public void setNota(Integer nota) {
        this.nota = nota;
    }

    public String getComentario() {
        return comentario;
    }

    public void setComentario(String comentario) {
        this.comentario = comentario;
    }

    public LocalDateTime getDataCriacao() {
        return dataCriacao;
    }

    public void setDataCriacao(LocalDateTime dataCriacao) {
        this.dataCriacao = dataCriacao;
    }

    @PrePersist
    void prePersist() {
        if (dataCriacao == null) dataCriacao = LocalDateTime.now();
    }
}
